components/infra-proxy-service/migrations/pipeline/utility.go

```diff
@@ -217,10 +223,75 @@ func openOrgFolder(org os.FileInfo, fileLocation string) OrgJson {
 	return orgJson
 }
 
-func createOrgStructForAction(orgId string, orgName string, ops ActionOps) Org {
-	return Org{
+func createOrgStructForAction(orgId string, orgName string, ops pipeline.ActionOps) pipeline.Org {
+	return pipeline.Org{
 		Name:      orgId,
 		FullName:  orgName,
 		ActionOps: ops,
 	}
 }
+
+func Unzip(ctx context.Context, mst storage.MigrationStorage, result pipeline.Result) (pipeline.Result, error) {
+	var fpath string
+	r, err := zip.OpenReader(result.Meta.ZipFile)
+	if err != nil {
+		log.Errorf("cannot open reader: %s.", err)
+		mst.FailedUnzip(ctx, result.Meta.MigrationID, result.Meta.ServerID, "cannot open zipfile", 0, 0, 0)
+		return result, err
+	}
+	defer r.Close()
+
+	for _, f := range r.File {
+		fpath = filepath.Join("", f.Name)
```

Comment:
will fpath will have the entire path for unzip folder?

Reply:
yes